d3servercheck.ps1

```diff
@@ -1,70 +1,52 @@
-function Write-Server-Result($tcpconns) {
+function Write-ServerResult($tcpconns) {
     if (!$tcpconns) {
         return
     }
-
-    foreach ($tcpconn in $tcpconns) {
-        Write-Host "$($tcpconn.RemoteAddress) (" -NoNewline
-        if ($is_good_ip.ContainsKey($tcpconn.RemoteAddress)) {
-            if ($is_good_ip[$tcpconn.RemoteAddress]) {
-                Write-Host "GOOD" -NoNewline
+    $Unique_Addresses = $tcpconns | Sort-object -property RemoteAddress -unique
+    write-host "Connections:"
+    foreach ($address in $Unique_Addresses) {
+        if ($is_good_ip.ContainsKey($address.RemoteAddress)) {
+            if ($is_good_ip[$address.RemoteAddress]) {
+                $Result = "GOOD"
             } else {
-                Write-Host "BAD" -NoNewline
+                $Result =  "BAD" 
             }
-        } else {
-            Write-Host "UNKNOWN" -NoNewline
         }
-
-        Write-Host ") " -NoNewline
+        else{
+            $Result = 'Unknown'
+        }
+        
+        $Message = '{0} : ({1})' -f $address.remoteaddress,$Result
+        Write-Host "$message"
     }
+    write-host "`r"
 }
 
 Write-Host "Starting d3servercheck"
 
 $csv = Import-Csv -Path servers.csv
 
+try {
+    $Diablo3_Process = get-process -name 'Diablo III64' -ErrorAction Stop
+}
+Catch{
+    write-error "Diablo 3 is not running. Exiting script..."
+    exit 1
+}
+
 $is_good_ip = @{}
 
 foreach ($ip in $csv) {
     $is_good_ip.Add($ip.server, $ip.status -eq "good")
 }
 
-$blizzard_server_ips_32 = @()
-$blizzard_server_ips_33 = @()
-$known_ips = $csv.server
-for ($j=0; $j -lt 256; $j++) {
-    $ip_32 = "37.244.32.$($j)"
-    $ip_33 = "37.244.33.$($j)"
-    if (!$is_good_ip.ContainsKey($ip_32)) {
-        $blizzard_server_ips_32 += $ip_32
-    }
-
-    if (!$is_good_ip.ContainsKey($ip_33)) {
-        $blizzard_server_ips_33 += $ip_33
-    }
-   
-}
-
 Write-Host "Read servers"
 Write-Host "Starting checks"
 
 while ($true) {
     Start-Sleep -s 5
 
```

Comment:
Check for D3 PID here (if feasible, IDK how expensive `pgrep` is on Windows, else increase sleep time?)